packages/s2-core/src/interaction/event-controller.ts

```diff
@@ -188,8 +188,20 @@ export class EventController {
     interaction.reset();
   }
 
-  private isMouseOnTheCanvasContainer(event: Event) {
+  private isMouseEvent(event: Event) {
```

Comment:
这里可以用 typescript 返回值类型推断限定

```suggestion
  private isMouseEvent(event: Event): event is MouseEvent {
```
下面 L204 就不需要了，在 if 块中，event 会直接被限定为 MouseEvent 了，试试？